src/components/Tag/Tabs/CompareImages.jsx

```diff
@@ -0,0 +1,154 @@
+import React, { useState, useMemo } from 'react';
+
+// components
+import { Button, Modal, Typography, Stack, Box } from '@mui/material';
+import makeStyles from '@mui/styles/makeStyles';
+
+import { api, endpoints } from 'api';
+import { host } from 'host';
+import ImageSelector from 'components/Tag/Tabs/ImageSelector';
+import VulnerabilitiyCard from 'components/Shared/VulnerabilityCard';
+import { isEmpty } from 'lodash';
+import { mapCVEInfo } from 'utilities/objectModels';
+
+const useStyles = makeStyles(() => ({
+  imagesInputBox: {
+    padding: '0.5rem',
+    marginBottom: '0.5rem'
+  },
+  input: {
+    color: '#464141',
+    '&::placeholder': {
+      opacity: '1'
+    }
+  },
+  searchInputBase: {
+    width: '90%',
+    paddingLeft: '1rem',
+    border: '1px solid black',
+    height: 40,
+    color: 'rgba(0, 0, 0, 0.6)'
+  },
+  compareImagesPopup: {
+    position: 'absolute',
+    top: '50%',
+    left: '50%',
+    transform: 'translate(-50%, -50%)',
+    backgroundColor: 'white',
+    border: '2px solid #000',
+    padding: '1rem'
+  },
+  compareButton: {
+    paddingLeft: '0.5rem'
+  }
+}));
+
+function CompareImages({ name, tag, platform }) {
+  const classes = useStyles();
+  const [open, setOpen] = useState(false);
+  const [minuend, setMinuend] = useState('');
+  const [subtrahend, setSubtrahend] = useState('');
+  const [cveData, setCVEData] = useState([]);
+  const handleOpen = () => setOpen(true);
+  const handleClose = () => setOpen(false);
+  const handleMinuendInput = (value) => {
+    setMinuend(value);
+  };
+  const handleSubtrahendInput = (value) => {
+    setSubtrahend(value);
+  };
+  const abortController = useMemo(() => new AbortController(), []);
+
+  const imageCVECompare = (minuend, subtrahend) => {
+    api
+      .get(
+        `${host()}${endpoints.cveDiffForImages(minuend, subtrahend, { pageNumber: 1, pageSize: 9 })}`,
+        abortController.signal
+      )
+      .then((diffResponse) => {
+        const cveListData = mapCVEInfo(diffResponse.data.data.CVEDiffListForImages.CVEList);
```

Comment:
we should have some checks here before passing data to mapCVEInfo, make sure the response actually contains data